pkg/driver/driver.go

```diff
@@ -94,40 +105,51 @@ type Node struct {
 	Platforms []specs.Platform
 }
 
-func Boot(ctx context.Context, d Driver, pw progress.Writer) (*client.Client, string, error) {
-	try := 0
-	rand.Seed(time.Now().UnixNano())
-	for {
-		info, err := d.Info(ctx)
+type BuilderClients struct {
+	// The builder on the chosen node/pod
+	ChosenNode NodeClient
+
+	// If multiple builders present, clients to the other builders (excluding the chosen pod)
+	OtherNodes []NodeClient
+}
+type NodeClient struct {
+	NodeName       string
+	ClusterAddr    string
+	BuildKitClient *client.Client
+}
+
+func Boot(ctx context.Context, d Driver, pw progress.Writer) (*BuilderClients, error) {
+	err := fmt.Errorf("timeout before starting")
+	var info *Info
+	var results *BuilderClients
+	for err != nil {
+		select {
+		case <-ctx.Done():
+			return nil, errors.Wrap(err, "timed out trying to bootstrap builder")
+		default:
+		}
+
+		info, err = d.Info(ctx)
 		if err != nil {
-			return nil, "", err
+			return nil, err
 		}
-		try++
+
 		if info.Status != Running {
-			if try > maxBootRetries {
-				return nil, "", errors.Errorf("failed to bootstrap builder in %d attempts (%s)", try, err)
-			}
 			if err = d.Bootstrap(ctx, func(s *client.SolveStatus) {
 				if pw != nil {
 					pw.Status() <- s
 				}
-			}); err != nil && (strings.Contains(err.Error(), "already exists") || strings.Contains(err.Error(), "not found")) {
-				// This most likely means another build is running in parallel
-				// Give it just enough time to finish creating resources then retry
-				time.Sleep(25 * time.Millisecond * time.Duration(1+rand.Int63n(39))) // 25 - 1000 ms
+			}); err != nil {
```

Comment:
This makes me a bit nervous. I get it might take a bit for the builders to simmer down, but just sleeping randomly for 1000 milliseconds feels a bit weird.

Reply:
That's fair.  Let me drop this down to a much smaller pause to get the racing CLIs skewed but not as noticeable to a human.